kasa/smartprotocol.py

```diff
@@ -46,14 +47,19 @@ def __init__(
         transport: Optional[BaseTransport] = None,
         credentials: Optional[Credentials] = None,
         timeout: Optional[int] = None,
+        port: Optional[int] = None,
     ) -> None:
-        super().__init__(host=host, port=self.DEFAULT_PORT)
+        super().__init__(
+            host=host,
+            port=port or self.DEFAULT_PORT,
+            timeout=timeout or self.DEFAULT_TIMEOUT,
+        )
 
-        self._credentials: Credentials = credentials or Credentials(
+        self.credentials: Credentials = credentials or Credentials(
             username="", password=""
         )
-        self._transport: BaseTransport = transport or AesTransport(
-            host, credentials=self._credentials, timeout=timeout
+        self.transport: BaseTransport = transport or AesTransport(
+            host, credentials=self.credentials, timeout=timeout, port=port
```

Comment:
If we contain the necessary information in the connectionparameters, most of this boilerplate would be unnecessary as the information would be accessible through the parameters object.

Reply:
Done